cmd/posix.go

```diff
@@ -571,19 +600,61 @@ func (s *posix) ReadFile(volume string, path string, offset int64, buf []byte) (
 		return 0, err
 	}
 
-	// Verify if its not a regular file, since subsequent Seek is undefined.
+	// Verify if its not a regular file, since subsequent Seek is
+	// undefined.
 	if !st.Mode().IsRegular() {
 		return 0, errIsNotRegular
 	}
 
-	// Seek to requested offset.
-	_, err = file.Seek(offset, os.SEEK_SET)
-	if err != nil {
-		return 0, err
+	// If empty hash algo or an expected hash string is empty we
+	// do not perform hash verification.
+	needToHash := !(algo == HashAlgo("") || expectedHash == "")
```

Comment:
Isn't it an error if the either the algo is set or the expectedHash is set (both not both)? Currently this would be silently ignored.